deploy/lib/createFunctions.js

```diff
@@ -1,31 +1,19 @@
 'use strict';
 
 const BbPromise = require('bluebird');
-const utils = require('../../shared/utils');
-const fs = require('fs');
 const path = require('path');
 
 module.exports = {
   createFunctions () {
     const createFunctionPromises = [];
 
     this.serverless.service.getAllFunctions().forEach((functionName) => {
-      const metaData = utils.getFunctionMetaData(functionName, this.provider.getParsedBindings(), this.serverless);
 
-      createFunctionPromises.push(this.provider.createZipObjectAndUploadFunction(functionName, metaData.entryPoint, metaData.handlerPath, metaData.params));
+      createFunctionPromises.push(this.provider.uploadFunction(functionName)
+        .then(() => this.provider.runKuduCommand('mv '+ path.join(functionName, functionName +'-function.json') +' '+ path.join(functionName, 'function.json'))));
     });
 
-    const packageJsonFilePath = path.join(this.serverless.config.servicePath, 'package.json');
-    let createFunctionsPromise = BbPromise.all(createFunctionPromises)
-                                .then(() => this.provider.syncTriggers())
-                                .then(() => this.provider.runKuduCommand('del package.json'));
-
-    if (fs.existsSync(packageJsonFilePath)) {
-      return createFunctionsPromise.then(() => this.provider.uploadPackageJson())
-            .then(() => this.provider.runKuduCommand('npm install --production'));
```

Comment:
Nope, `npm install` will never run automatically. But we could think of adding some flag, so the user can explicitly run it if he wants to.

Not sure if we need to validate anything this is developers responsibility, aws provider works that way. You have only dependencies what you packed.